WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListItemUiStateHelper.kt

```diff
@@ -314,7 +333,9 @@ class PostListItemUiStateHelper @Inject constructor(private val appPrefsWrapper:
         statsSupported: Boolean
     ): List<PostListButtonType> {
         val canRetryUpload = uploadUiState is PostUploadUiState.UploadFailed
-        val canCancelPendingAutoUpload = uploadUiState is UploadWaitingForConnection && postStatus != DRAFT
+        val canCancelPendingAutoUpload = (uploadUiState is UploadWaitingForConnection ||
```

Comment:
We want to show "cancel" even when there is a failed upload in case the post is eligible for auto-upload.